pkg/config/parameters.go

```diff
@@ -321,6 +321,10 @@ type TLSParameters struct {
 	// by advanced users. Note that these will be ignored when TLS 1.3 is in
 	// use.
 	CipherSuites TLSCiphers `yaml:"cipher-suites,omitempty"`
+
+	// EnableUpstreamTLS will apply MinimumProtocolVersion and CipherSuites on every
+	// Envoy -> UpstreamCluster communication and Envoy -> ExternalServiceCluster communication
+	EnableUpstreamTLS bool `yaml:"enable-upstream-tls"`
```

Comment:
is this the API we want @projectcontour/maintainers? this means you *must* set ciphers if you want to propagate these global tls settings to upstream connections, rather than being able to just set min tls version or ciphers and have that propagate

also could do with a little bit more specific naming of this field, maybe something that conveys more that the TLS settings in this section are going to be applied to upstream connections

Reply:
might want to change the godoc on the CipherSuites field too since that mentions Envoy listeners speciifcally, and now we want to apply this to upstream connections

Reply:
Yes, I think that users should just be able to enable upstream TLS and then have the upstream use the same values as whatever the downstream is configured to, by default. If people want to customise, they can.

Reply:
Although it would result in less streamlined API, I wonder if it would be safer to have full configurability. There might be different needs for downstream vs upstream. Maybe one could have strict TLS 1.3 only policy for cluster-internal traffic but less restrictive towards external clients (although the current API without upstream controls, is basically the opposite). Or for practical reasons some weaker TLS 1.2 ciphers might need to be enabled for external interface but are unnecessary for internal.